packages/react-native-audio-api/common/cpp/audioapi/core/sources/AudioBufferSourceNode.h

```diff
@@ -22,41 +23,69 @@ class AudioBufferSourceNode : public AudioScheduledSourceNode {
   [[nodiscard]] double getLoopEnd() const;
   [[nodiscard]] std::shared_ptr<AudioParam> getDetuneParam() const;
   [[nodiscard]] std::shared_ptr<AudioParam> getPlaybackRateParam() const;
+  [[nodiscard]] std::shared_ptr<AudioParam> getSemitonesParam() const;
   [[nodiscard]] std::shared_ptr<AudioBuffer> getBuffer() const;
+  [[nodiscard]] std::string getTimeStretchType() const;
 
   void setLoop(bool loop);
   void setLoopStart(double loopStart);
   void setLoopEnd(double loopEnd);
   void setBuffer(const std::shared_ptr<AudioBuffer> &buffer);
+  void setTimeStretchType(const std::string &type);
 
   void start(double when, double offset, double duration = -1);
 
  protected:
   std::mutex &getBufferLock();
   void processNode(const std::shared_ptr<AudioBus>& processingBus, int framesToProcess) override;
+  double getStopTime() const override;
 
  private:
+  static TimeStretchType fromString(const std::string &type) {
+    std::string lowerType = type;
+    std::transform(
+        lowerType.begin(), lowerType.end(), lowerType.begin(), ::tolower);
+
+    if (lowerType == "linear")
+      return TimeStretchType::LINEAR;
+    if (lowerType == "speech-music")
+        return TimeStretchType::SPEECH_MUSIC;
+
+    throw std::invalid_argument("Unknown time stretch type: " + type);
+  }
+
+  static std::string toString(TimeStretchType type) {
+    switch (type) {
+      case TimeStretchType::LINEAR:
+        return "linear";
+      case TimeStretchType::SPEECH_MUSIC:
+        return "speech-music";
+    }
+
+    throw std::invalid_argument("Unknown time stretch type");
+  }
+
   // Looping related properties
   bool loop_;
   double loopStart_;
   double loopEnd_;
   std::mutex bufferLock_;
 
-  // playback rate aka pitch change params
+  // time-stretching and pitch-shifting
+  TimeStretchType timeStretchType_;
+
+  // k-rate params
   std::shared_ptr<AudioParam> detuneParam_;
   std::shared_ptr<AudioParam> playbackRateParam_;
+  std::shared_ptr<AudioParam> semitones_;
```

Comment:
Potentially `semitones` and `detune` could be the same parameter, as they represent same thing more or less (-octave, +octave)